cloudplatform/connectivity-destination-service/src/main/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationKeyStoreExtractor.java

```diff
@@ -254,6 +263,29 @@ private static KeyStore retrieveExistingKeyStore(
         }
     }
 
+    @Nonnull
+    static KeyStore createNewKeyStoreFromPem( @Nonnull final String data, @Nullable final String password )
+    {
+        try {
+            final String s = new String(Base64.getDecoder().decode(data), StandardCharsets.UTF_8).trim();
+
+            final Pattern p = Pattern.compile("[-]+BEGIN CERTIFICATE[-]+.*[-]+END CERTIFICATE[-]+", Pattern.DOTALL);
```

Comment:
Can be simplified to:

```suggestion
            final Pattern p = Pattern.compile("-+BEGIN CERTIFICATE-+.*-+END CERTIFICATE-+", Pattern.DOTALL);
```